server/src/main/java/io/crate/types/TimeTZType.java

```diff
@@ -0,0 +1,113 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+import static io.crate.types.TimeTZParser.timeTZOf;
+import static io.crate.types.TimeTZParser.exceptionForInvalidLiteral;
+
+public final class TimeTZType extends DataType<TimeTZ> implements FixedWidthType, Streamer<TimeTZ> {
+
+    public static final int ID = 20;
+    public static final int TYPE_LEN = 12;
+    public static final String NAME = "time with time zone";
+    public static final TimeTZType INSTANCE = new TimeTZType();
+
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public Precedence precedence() {
+        return Precedence.TIMETZ;
+    }
+
+    @Override
+    public Streamer<TimeTZ> streamer() {
+        return this;
+    }
+
+    @Override
+    public int compare(TimeTZ val1, TimeTZ val2) {
+        return val1.compareTo(val2);
+    }
+
+    @Override
+    public TimeTZ readValueFrom(StreamInput in) throws IOException {
+        if (in.readBoolean()) {
+            return null;
+        }
+        return new TimeTZ(in.readLong(), in.readInt());
+    }
+
+    @Override
+    public void writeValueTo(StreamOutput out, TimeTZ tz) throws IOException {
+        out.writeBoolean(tz == null);
+        if (tz != null) {
+            out.writeLong(tz.getMicrosFromMidnight());
+            out.writeInt(tz.getSecondsFromUTC());
+        }
+    }
+
+    @Override
+    public int fixedSize() {
+        return TYPE_LEN;
+    }
+
+    @Override
+    public TimeTZ implicitCast(Object value) {
+        if (value instanceof String) {
+            try {
+                return TimeTZParser.parse((String) value);
+            } catch (IllegalArgumentException e0) {
+                try {
+                    return timeTZOf(
+                        TimeTZType.class.getSimpleName(),
+                        Long.valueOf((String) value));
+                } catch (NumberFormatException e1) {
+                    throw exceptionForInvalidLiteral(value);
+                }
+            }
+        }
+        throw exceptionForInvalidLiteral(value);
+    }
+
+    @Override
+    public TimeTZ value(Object value) {
+        if (value == null) {
+            return null;
+        }
+        return value instanceof TimeTZ ? (TimeTZ) value : implicitCast(value);
```

Comment:
`DataType#value` must not call `DataType#implicitCast` and perform the casting.
You would probably also have to adjust tests accordingly. 

I'll copy-paste jordi's comment from the slack discussion regarding refactoring `DataType` and splitting
`DataType#value` for a better context.

```
Basically what I'm proposing is to split up value into multiple use cases because
right now it is a bit overloaded:
- cast, with a comment that this is only to be used via the cast function
 CastFunction, supports explicit/try/implicit
- valueForInsert - for the new implicit whitespace truncation case
- santizeType - for the cases where we currently use dataType#value to fix
a type read from source or so where an integer might have been stored as long
(This is mostly in the reference resolvers / column expression implementations)
``` 

For the `timetz` it seems that it'd be enough to implement the `implicitCast`, which is done already and keep the `DataType#value` as it was suggested in the previous comment.